### PR TITLE
Node build environment

### PR DESCRIPTION
**Pull request by Burke-Daniel**
This PR creates our shiny new build environment for Node (finally!). We have decided to not use STM32CubeIDE, in favour of doing our own CMake, it works better cross-platform and then everyone can use whatever development environment they please. See the new README section for getting set up to build.

### CODE REVIEW

File: Examples/Blink/Blink.ioc [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ProjectManager.HalAssertFull=false
+RCC.RTCHSEDivFreq_Value=250000
+PA2.Signal=USART2_TX
+PA13.GPIO_Label=TMS
+ProjectManager.ProjectName=Blink
+ProjectManager.UnderRoot=false
+RCC.TIM1Freq_Value=72000000
+PA13.GPIOParameters=GPIO_Label
+RCC.PLLMCOFreq_Value=72000000
+ProjectManager.CoupleFile=false
```
Comment (Akashf) [inline Examples/Blink/Blink.ioc:123]: The "generate peripheral initialization as a pai of '.c/.h' files per peripheral" option must be set as the generated code will go to main.c, which is being deleted

**Review by Akashf — COMMENTED** [review]
I was able to build everything while following the instructions. Probably should add .DS_Store to the .gitignore for our resedential mac user.

File: README.md [hunk 2]
```diff
@@ -13,3 +13,23 @@ Prerequisite: [Node.js(LTS version will be fine) and npm](https://docs.npmjs.com
 4. When you in the angular-UI folder, run `npm install`
 5. To build and serves the app locally, run `ng serve`. If it's build successfully, there should be a link that look like this: http://localhost:4200/. 
 6. Navigate to the link in your browser, and your should be able to see the app.
+
+## Node (embedded) setup Guide
+Prerequisites:
+ - [STM32CubeMx](https://www.st.com/en/development-tools/stm32cubemx.html)
```
Comment (Burke-Daniel) [inline README.md:19]: Reminder to add cubeMx version to README

File: README.md [hunk 3]
```diff
@@ -13,3 +13,23 @@ Prerequisite: [Node.js(LTS version will be fine) and npm](https://docs.npmjs.com
...
+ - Windows Users will need [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
+ - The following packages (should be able to install from package manager):
```
Comment (Burke-Daniel) [inline README.md:21]: add example of command for installing

File: Examples/Blink/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,106 @@
+cmake_minimum_required(VERSION 3.15.3)
```
Comment (Burke-Daniel) [inline Examples/Blink/CMakeLists.txt:1]: change to 3.10.2

File: README.md [hunk 5]
```diff
@@ -13,3 +13,23 @@ Prerequisite: [Node.js(LTS version will be fine) and npm](https://docs.npmjs.com
...
+   - cmake
```
Comment (Burke-Daniel) [inline README.md:22]: specify minimum required version

File: README.md [hunk 6]
```diff
@@ -13,3 +13,23 @@ Prerequisite: [Node.js(LTS version will be fine) and npm](https://docs.npmjs.com
...
+   - build-essential
+   - gcc-arm-none-eabi
```
Comment (Burke-Daniel) [inline README.md:24]: specify version

File: README.md (hunk 5 above)
Comment (colton-smith) [reply README.md:22]: 3.15.3 should be fine - We can do 3.13 for flight computer or whatever the default raspbian CMake version is

**Review by colton-smith — APPROVED** [review]
Went through all of the motions, ran into some problems specific to Ubuntu 18.04 as we discussed, the default CMake version is lower. I went ahead and added the updated readme to handle updating CMake.

Other than that it all looks good! Only thing is we have to figure out how this is gonna be setup for ~5 nodes (whats shared and whats not between each Node) . I guess this will come in a later PR? 

Other than that - GTM! I would like someone to run through it and throw their review down (particularly a mac user) before you merge this though.

File: Examples/Blink/Blink.ioc (hunk 1 above)
Comment (Burke-Daniel) [reply Examples/Blink/Blink.ioc]: Good point, fixed

File: README.md (hunk 5 above)
Comment (Burke-Daniel) [reply README.md]: Sounds good

**Comment by Burke-Daniel** [comment]
@adrwstn Would like for you to take a full run through this setup on mac and report back if you run into any issues, after that will go ahead and merge

**Review by adrwstn — APPROVED** [review]
Build setup working on Mac OS 11, good to merge

**Pull request merged**